core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -417,7 +418,7 @@ final class ShuffleBlockFetcherIterator(
         numBlocksToFetch += mergedBlockInfos.size
         localBlocks ++= mergedBlockInfos.map(info => (info.blockId, info.mapIndex))
         localBlockBytes += mergedBlockInfos.map(_.size).sum
-      } else if (blockManager.hostLocalDirManager.isDefined &&
+      } else if (doLocalDiskRead && blockManager.hostLocalDirManager.isDefined &&
```

Comment:
Hmm, if `SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED` is disabled, I think `BlockManager` won't have `hostLocalDirManager` (i.e., it is `None`), except push-based shuffle is enabled.

But I think push-based shuffle is only enabled under YARN mode. For K8s, it is disabled.

How does it enters this branch if `blockManager.hostLocalDirManager.isDefined` is false?

Reply:
Ah, I think you are right! Looks like disabling `SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED` is good enough on K8s. Let me do some more testing.

Reply:
Did a quick test by disabling `SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED`, looks good. Will close this PR.